### ISSUE TITLE
Scala runner doesn't do executable jars

### ISSUE THREAD

**Issue by som-snytt** [lampepfl/dotty#12569]
## Compiler version

3.0.0

## Minimized code

```Scala
➜  snips sdk use scala 2.13.6

Using scala version 2.13.6 in this shell.
➜  snips scalac -d hi.jar hi.scala
➜  snips scala hi.jar
hello, world
➜  snips sdk use scala 3.0.0

Using scala version 3.0.0 in this shell.
➜  snips scalac -d hi.jar hi.scala
➜  snips scala hi.jar
Error: Could not find or load main class hi.jar
Caused by: java.lang.ClassNotFoundException: hi.jar
➜  snips scala -classpath hi.jar Main
hello, world
```

## Output

```scala
Error: Could not find or load main class hi.jar
```

## Expectation

Scala runner creates and runs an executable jar. It looks like it doesn't create `META-INF/MANIFEST.MF`.

**Comment by okarmusk** [comment]
I've got the same issue when run `scala 3` from binary.
Bug is visible even if jar is build from `sbt package` where sbt is in version 1.5.2. Please check issue what has been added by @adpi2

**Comment by okarmusk** [comment]
I built the same source code with the same sbt version but I compile and run it with `scala 2.13.6`, it works.
Issue exists only in `scala 3`.

**Comment by som-snytt** [comment]
I'll take a swing at this, since it's such a useful niche feature. By swing, I mean a bunt down the 3rd base line.

**Comment by dwijnand** [comment]
Overlaps with #13081, so either make sure to coordinate with @BarkingBad.

**Comment by BarkingBad** [comment]
I tested it @anatoliykmetyuk and compiling files using `scalac -d hi.jar hi.scala` doesn't provide `Manifest` file. The way to get runnable jar is to run it by `scala -save hi.scala` and then run it as `scala hi.jar`

Should we introduce change to `scalac` that would make Manifest file?

**Comment by som-snytt** [comment]
Scala 2 distinguished
```
  -Xmain-class <path>              Class for manifest's Main-Class entry (only useful with -d <jar>)
  -Xscript <object>                Treat the source file as a script and wrap it in a main method.
```
and REPL
```
Other startup options:

 -i <file>    preload <file> before starting the REPL
 -I <file>    preload <file>, enforcing line-by-line interpretation
 -e <string>  execute <string> as if it were in a source file
 -save        save the compiled script in a jar for future use
```

**Comment by anatoliykmetyuk** [comment]
> Should we introduce change to scalac that would make Manifest file?

I believe so. In any case this shouldn't be a `-save` option to produce a manifest for a runnable jar, as "save" isn't very related to "runnable".

**Comment by BarkingBad** [comment]
I guess bringing back `-Xmain-class` should be easy.
However, I don't know if `-Xscript` should be in the scope of this task and whether we need it at all in the compiler, we can run scripts by running `scala <path-to-script>`

**Comment by anatoliykmetyuk** [comment]
Yes, we don't need `-Xscript` anymore since we have a better, more seamless, scripting support in Scala 3.

**Comment by anatoliykmetyuk** [comment]
Yes, we discussed whether we should go Python-style or tolerate `@main` in scripts earlier. The argument for `@main` was not only simplicity but also the ability to accept command arguments in a natural way. Another argument was that scripts and programs ideally should be the same thing; we should strive to avoid differentiating between the two. Overall we got the consensus that `@main` in scripts is fine.
